crates/solver/src/settlement_submission.rs

```diff
@@ -311,6 +313,20 @@ impl SolutionSubmitter {
         };
 
         let strategy_args = strategy.strategy_args().expect("unreachable code executed");
+
+        // No extra tip required if there is no revert risk
+        let (additional_tip_percentage_of_max_fee, max_additional_tip) =
+            if shared::gas_price_estimation::is_mainnet(&network_id)
+                && settlement.revertable() == Revertable::NoRisk
+            {
+                (
+                    strategy_args.additional_tip_percentage_of_max_fee,
+                    strategy_args.max_additional_tip,
+                )
+            } else {
+                (0., 0.)
+            };
```

Comment:
Might be worth to log this, it would have been easier to debug that way.

If you do, you may want to move the instrumentation 👇 so that this has a span tied to the submitter.